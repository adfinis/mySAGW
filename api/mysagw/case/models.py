from django.db import models

from mysagw.identity.models import Identity
from mysagw.models import HistoricalModel, UUIDModel


class CaseAccess(UUIDModel, HistoricalModel):
    case_id = models.UUIDField()
    identity = models.ForeignKey(
        Identity,
        related_name="cases",
        on_delete=models.CASCADE,
        null=True,
    )
    email = models.EmailField(null=True)

    class Meta:
        unique_together = [["case_id", "email"], ["case_id", "identity"]]
