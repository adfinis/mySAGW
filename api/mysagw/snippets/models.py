from django.db import models
from localized_fields.fields import LocalizedTextField

from mysagw.models import HistoricalModel, UUIDModel


class Snippet(UUIDModel, HistoricalModel):
    title = models.CharField(max_length=255)
    body = LocalizedTextField(blank=False, null=False, required=False)
    archived = models.BooleanField(default=False)

    class Meta:
        ordering = ("title",)
