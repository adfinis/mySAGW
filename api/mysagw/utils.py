def build_url(*fragments, **options):
    sep = "/"
    url = sep.join([fragment.strip(sep) for fragment in fragments])

    if options.get("trailing", False):
        url += sep

    return url


def format_currency(value, currency):
    if currency and (isinstance(value, (float, int))):
        value = f"{currency.upper()} {value:_.2f}".replace(".00", ".-").replace(
            "_",
            "'",
        )
    return value
