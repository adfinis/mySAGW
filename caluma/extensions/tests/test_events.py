import pytest
from django.utils import timezone

from caluma.caluma_form.models import Question
from caluma.caluma_workflow.api import (
    complete_work_item,
    redo_work_item,
    reopen_case,
    skip_work_item,
    start_case,
)
from caluma.caluma_workflow.models import Case, Workflow, WorkItem

from ..settings import settings


def test_work_item_set_assigned_user(
    db, caluma_data, case_access_event_mock, document_review_case
):
    assert document_review_case.work_items.get(
        task_id="submit-document"
    ).assigned_users == ["test"]


def test_work_item_create_circulation(
    db, caluma_data, case_access_event_mock, circulation
):
    assert circulation.workflow_id == "circulation"


def test_work_item_finish_circulation(
    db, caluma_data, user, case_access_event_mock, circulation
):
    complete_work_item(circulation.work_items.get(task_id="finish-circulation"), user)

    assert (
        circulation.work_items.get(task_id="invite-to-circulation").status
        == WorkItem.STATUS_CANCELED
    )


def test_work_item_additional_data(
    db, caluma_data, user, case_access_event_mock, circulation
):
    Question.objects.filter(formquestion__form__pk="additional-data-form").update(
        is_required="false"
    )

    case = circulation.parent_work_item.case

    skip_work_item(case.work_items.get(task_id="circulation"), user)

    case.work_items.get(task_id="decision-and-credit").document.answers.create(
        question_id="decision-and-credit-decision",
        value="decision-and-credit-decision-additional-data",
    )

    skip_work_item(case.work_items.get(task_id="decision-and-credit"), user)

    assert (
        case.work_items.get(task_id="additional-data").status == WorkItem.STATUS_READY
    )

    complete_work_item(case.work_items.get(task_id="additional-data"), user)

    assert (
        case.work_items.get(task_id="advance-credits").status == WorkItem.STATUS_READY
    )
    assert (
        case.work_items.get(task_id="additional-data-form").status
        == WorkItem.STATUS_SUSPENDED
    )


def test_work_item_define_amount(
    db, caluma_data, user, case_access_event_mock, circulation, send_mail_mock
):
    work_items = circulation.parent_work_item.case.work_items

    skip_work_item(work_items.get(task_id="circulation"), user)
    work_items.get(task_id="decision-and-credit").document.answers.create(
        question_id="decision-and-credit-decision",
        value="decision-and-credit-decision-additional-data",
    )
    complete_work_item(work_items.get(task_id="decision-and-credit"), user)
    complete_work_item(work_items.get(task_id="additional-data"), user)
    work_items.get(task_id="define-amount").document.answers.create(
        question_id="define-amount-decision",
        value="define-amount-decision-reject",
    )
    skip_work_item(work_items.get(task_id="define-amount"), user)

    assert (
        work_items.get(task_id="additional-data-form").status
        == WorkItem.STATUS_SUSPENDED
    )

    complete_work_item(
        work_items.get(task_id="additional-data", status=WorkItem.STATUS_READY),
        user,
    )

    answers = work_items.get(task_id="additional-data-form").document.answers

    answers.create(
        question_id="define-amount-decision",
        value="define-amount-decision-continue",
    )
    answers.create(
        question_id="additional-data-adresse",
        value="Foo",
    )
    answers.create(
        question_id="additional-data-bank",
        value="Foo",
    )
    answers.create(
        question_id="additional-data-iban",
        value="Foo",
    )
    answers.create(
        question_id="additional-data-name",
        value="Foo",
    )

    define_amount = work_items.get(
        task_id="define-amount", status=WorkItem.STATUS_READY
    )
    define_amount.document.answers.create(
        question_id="define-amount-decision",
        value="define-amount-decision-continue",
    )
    complete_work_item(define_amount, user)

    assert work_items.get(task_id="complete-document").status == WorkItem.STATUS_READY
    assert (
        work_items.get(task_id="additional-data-form").status
        == WorkItem.STATUS_COMPLETED
    )
    assert work_items.get(task_id="advance-credits").status == WorkItem.STATUS_COMPLETED
    assert circulation.parent_work_item.case.status == Case.STATUS_RUNNING


def test_work_item_invite_to_circulation(
    db, caluma_data, user, case_access_event_mock, circulation
):
    complete_work_item(
        circulation.work_items.get(task_id="invite-to-circulation"),
        user,
        {
            "assign_users": [
                {"idpId": "ab", "name": "AB", "email": "a@b.c"},
                {"idpId": "cd", "name": "CD", "email": "c@d.e"},
            ]
        },
    )

    decisions = circulation.work_items.filter(task_id="circulation-decision")
    assert decisions.count() == 2
    assert list(
        map(lambda v: v["assigned_users"][0], decisions.values("assigned_users"))
    ) == [
        "ab",
        "cd",
    ]


def test_case_complete_circulation(
    db, caluma_data, user, case_access_event_mock, circulation
):
    complete_work_item(circulation.work_items.get(task_id="finish-circulation"), user)

    assert circulation.parent_work_item.status == WorkItem.STATUS_COMPLETED


@pytest.mark.parametrize(
    "same_year,first", [(True, False), (False, False), (False, True)]
)
def test_case_number(
    db,
    caluma_data,
    case_access_event_mock,
    user,
    form_question_factory,
    answer_factory,
    same_year,
    first,
):
    question = Question.objects.get(pk="dossier-nr")
    form_question = form_question_factory(question=question)
    old_year = year = timezone.now().year
    if not same_year:
        old_year = old_year - 1

    if not first:
        for i in range(1, 4):
            answer_factory(question=form_question.question, value=f"{old_year}-000{i}")

    expected_no = f"{year}-0004"
    if not same_year:
        expected_no = f"{year}-0001"

    case = start_case(
        workflow=Workflow.objects.get(pk="document-review"),
        form=form_question.form,
        user=user,
    )

    assert case.document.answers.get(question=question).value == expected_no


def test_case_status(
    db,
    caluma_data,
    case_access_event_mock,
    document_review_case,
    user,
):
    case = document_review_case
    assert case.meta["status"] == "submit"

    skip_work_item(case.work_items.get(task_id="submit-document"), user)
    assert case.meta["status"] == "audit"

    case.work_items.get(task_id="review-document").document.answers.create(
        question_id="review-document-decision",
        value="review-document-decision-continue",
    )
    skip_work_item(case.work_items.get(task_id="review-document"), user)
    assert case.meta["status"] == "audit"

    skip_work_item(case.work_items.get(task_id="circulation"), user)
    assert case.meta["status"] == "audit"

    case.work_items.get(task_id="decision-and-credit").document.answers.create(
        question_id="decision-and-credit-decision",
        value="decision-and-credit-decision-additional-data",
    )
    skip_work_item(case.work_items.get(task_id="decision-and-credit"), user)
    assert case.meta["status"] == "submit-receipts"

    skip_work_item(case.work_items.get(task_id="additional-data"), user)
    skip_work_item(case.work_items.get(task_id="additional-data-form"), user)
    assert case.meta["status"] == "decision"

    skip_work_item(case.work_items.get(task_id="advance-credits"), user)
    assert case.meta["status"] == "decision"

    case.work_items.get(task_id="define-amount").document.answers.create(
        question_id="define-amount-decision",
        value="define-amount-decision-continue",
    )
    skip_work_item(case.work_items.get(task_id="define-amount"), user)
    assert case.meta["status"] == "decision"

    skip_work_item(case.work_items.get(task_id="complete-document"), user)
    assert case.meta["status"] == "complete"

    reopen_case(case, [case.work_items.get(task_id="complete-document")], user)
    assert case.meta["status"] == "decision"


@pytest.mark.parametrize("lang", ["de", "en", "fr"])
@pytest.mark.parametrize(
    "task_slug, revise",
    [
        ("revise-document", False),
        ("additional-data", False),
        ("additional-data", True),
        ("complete-document", False),
        ("decision-and-credit", False),
    ],
)
def test_send_work_item_mail(
    db,
    user,
    caluma_data,
    case_access_event_mock,
    document_review_case,
    mailoutbox,
    task_slug,
    revise,
    lang,
    snapshot,
    mocker,
):
    if lang != "de":
        mock = mocker.patch("caluma.extensions.events.work_item.get_users_for_case")
        mock.return_value = [
            {
                "idp-id": "267796f2-ae48-4235-93d7-bf26e8ba66bb",
                "first-name": "Winston",
                "last-name": "Smith",
                "email": "test-send@example.com",
                "language": lang,
            }
        ]
    case = document_review_case

    review_document_decision_answer = "review-document-decision-continue"

    if task_slug == "revise-document":
        review_document_decision_answer = "review-document-decision-reject"

    case.document.answers.create(
        question_id="dossier-nr",
        value="1984-0023",
    )

    skip_work_item(case.work_items.get(task_id="submit-document"), user)
    review_work_item = case.work_items.get(task_id="review-document")
    review_work_item.document.answers.create(
        question_id="review-document-decision",
        value=review_document_decision_answer,
    )
    complete_work_item(review_work_item, user)

    if task_slug == "decision-and-credit":
        skip_work_item(case.work_items.get(task_id="circulation"), user)
        decision_credit_work_item = case.work_items.get(task_id="decision-and-credit")
        decision_credit_work_item.document.answers.create(
            question_id="decision-and-credit-decision",
            value="decision-and-credit-decision-close",
        )
        complete_work_item(decision_credit_work_item, user)

    if task_slug in ["additional-data", "complete-document"]:
        skip_work_item(case.work_items.get(task_id="circulation"), user)
        decision_credit_work_item = case.work_items.get(task_id="decision-and-credit")
        decision_credit_work_item.document.answers.create(
            question_id="decision-and-credit-decision",
            value="decision-and-credit-decision-additional-data",
        )
        decision_credit_work_item.document.answers.create(
            question_id="gesprochener-rahmenkredit",
            value=23000.0,
        )
        complete_work_item(decision_credit_work_item, user)

        if revise:
            complete_work_item(case.work_items.get(task_id="additional-data"), user)
            define_amount_work_item = case.work_items.get(task_id="define-amount")
            define_amount_work_item.document.answers.create(
                question_id="define-amount-decision",
                value="define-amount-decision-reject",
            )
            complete_work_item(define_amount_work_item, user)

    if task_slug == "complete-document":
        additional_data_work_item = case.work_items.get(task_id="additional-data")
        additional_data_form_work_item = case.work_items.get(
            task_id="additional-data-form"
        )
        additional_data_form_work_item.document.answers.create(
            question_id="additional-data-adresse",
            value="street",
        )
        additional_data_form_work_item.document.answers.create(
            question_id="additional-data-bank",
            value="bank",
        )
        additional_data_form_work_item.document.answers.create(
            question_id="additional-data-iban",
            value="iban",
        )
        additional_data_form_work_item.document.answers.create(
            question_id="additional-data-name",
            value="name",
        )
        complete_work_item(additional_data_form_work_item, user)
        complete_work_item(additional_data_work_item, user)

        define_amount_work_item = case.work_items.get(task_id="define-amount")
        define_amount_work_item.document.answers.create(
            question_id="define-amount-amount-float",
            value=23000.0,
        )
        define_amount_work_item.document.answers.create(
            question_id="define-amount-decision",
            value="define-amount-decision-continue",
        )
        complete_work_item(define_amount_work_item, user)

    expected_mails = 1
    if task_slug == "complete-document" or revise:
        expected_mails = 2

    assert len(mailoutbox) == expected_mails
    mail = mailoutbox[expected_mails - 1]
    assert mail.from_email == settings.MAILING_SENDER
    assert mail.to == ["test-send@example.com"]
    snapshot.assert_match(mail.subject)
    snapshot.assert_match(mail.body.replace(str(case.pk), "my-case-id"))


def test_access_control(
    db,
    caluma_data,
    user,
    identities_mock,
    form_question_factory,
    get_token_mock,
    case_access_create_request_mock,
):
    form_question = form_question_factory()

    start_case(
        workflow=Workflow.objects.get(pk="document-review"),
        form=form_question.form,
        user=user,
    )
    assert case_access_create_request_mock.called


def test_redo_circulation(
    db,
    caluma_data,
    user,
    case_access_event_mock,
    circulation,
    work_item_factory,
):
    Question.objects.filter(formquestion__form__pk="additional-data-form").update(
        is_required="false"
    )

    circ_work_item = circulation.parent_work_item

    work_item_factory(
        case=circulation, task_id="circulation-decision", status=WorkItem.STATUS_READY
    )
    skip_work_item(circ_work_item.case.work_items.get(task_id="circulation"), user)

    assert (
        circ_work_item.child_case.work_items.get(task_id="finish-circulation").status
        == WorkItem.STATUS_CANCELED
    )
    assert circ_work_item.child_case.work_items.count() == 3

    redo_work_item(circ_work_item.case.work_items.get(task_id="circulation"), user)

    circ_work_item.refresh_from_db()
    assert (
        circ_work_item.child_case.work_items.get(task_id="finish-circulation").status
        == WorkItem.STATUS_READY
    )
    assert circ_work_item.child_case.work_items.count() == 2


def test_redo_review_document(
    db,
    caluma_data,
    user,
    case_access_event_mock,
    circulation,
    work_item_factory,
):
    Question.objects.filter(formquestion__form__pk="additional-data-form").update(
        is_required="false"
    )

    circ_work_item = circulation.parent_work_item

    previous_case = circ_work_item.child_case.pk
    work_item_factory(
        case=circulation, task_id="circulation-decision", status=WorkItem.STATUS_READY
    )
    assert circ_work_item.child_case.work_items.count() == 3

    redo_work_item(circ_work_item.case.work_items.get(task_id="review-document"), user)

    circ_work_item.refresh_from_db()
    new_case = circ_work_item.child_case.pk
    assert previous_case != new_case
    assert circ_work_item.child_case.work_items.count() == 2


@pytest.mark.parametrize(
    "decision,expected_work_item_task",
    [
        (
            "define-amount",
            "define-amount",
        ),
        (
            "additional-data",
            "advance-credits",
        ),
    ],
)
def test_redo_define_amount(
    db,
    caluma_data,
    user,
    case_access_event_mock,
    circulation,
    decision,
    expected_work_item_task,
    send_mail_mock,
):
    Question.objects.filter(formquestion__form__pk="additional-data-form").update(
        is_required="false"
    )

    case = circulation.parent_work_item.case

    skip_work_item(case.work_items.get(task_id="circulation"), user)

    case.work_items.get(task_id="decision-and-credit").document.answers.create(
        question_id="decision-and-credit-decision",
        value=f"decision-and-credit-decision-{decision}",
    )

    complete_work_item(case.work_items.get(task_id="decision-and-credit"), user)

    if decision != "define-amount":
        complete_work_item(case.work_items.get(task_id="additional-data"), user)

    case.work_items.get(task_id="define-amount").document.answers.create(
        question_id="define-amount-decision", value="define-amount-decision-continue"
    )

    complete_work_item(case.work_items.get(task_id="define-amount"), user)

    redo_work_item(case.work_items.get(task_id="define-amount"), user)

    assert case.status == Case.STATUS_RUNNING
    assert case.work_items.filter(
        task_id=expected_work_item_task, status=WorkItem.STATUS_READY
    ).exists()
