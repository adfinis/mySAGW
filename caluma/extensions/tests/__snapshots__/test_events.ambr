# name: test_send_work_item_mail[additional-data-False-de]
  'Neue Informationen zu 1984-0023'
# ---
# name: test_send_work_item_mail[additional-data-False-de].1
  '''
  Guten Tag Winston Smith
  
  Wir haben Ihren Antrag geprüft und freuen uns, Ihnen mitteilen zu können, dass wir für das unten genannte Projekt einen Rahmenkredit von CHF 23'000.- gesprochen haben.
  
  Bei Bedarf können Sie Ihren Antrag und die Kostenzusprache als PDF unter der Rubrik «Download» herunterladen.
  
  Referenz: 1984-0023
  Über den nachfolgenden Link können Sie direkt auf Ihr Dokument zugreifen, wo Sie die Angaben zu den noch benötigten Informationen finden:
  https://mysagw.local/cases/my-case-id
  
  Sollten Sie Fragen haben, können Sie uns gerne über sagw@sagw.ch, die Telefonnummer +41 31 306 92 50 oder unter www.sagw.ch/kontakt-mysagw kontaktieren. Gerne bitten wir Sie, bei einer Frage immer die Referenznummer anzugeben.
  
  Freundliche Grüsse
  
  Ihr SAGW-Team
  
  ******
  
  Dies ist eine automatisch generierte Email, bitte antworten Sie nicht auf diese Nachricht. Die Kontaktangaben der SAGW können Sie der nachfolgenden Seite entnehmen:
  www.sagw.ch/kontakt-mysagw
  
  '''
# ---
# name: test_send_work_item_mail[additional-data-False-en]
  'New information concerning 1984-0023'
# ---
# name: test_send_work_item_mail[additional-data-False-en].1
  '''
  Dear Winston Smith
  
  After a careful review your application we are pleased to inform you that a framework credit of CHF 23'000.- has been approved for the project mentioned below.
  
  You can download your application and the approval of the framework credit as PDF files under the heading «Download».
  
  Reference: 1984-0023
  You can access your document via the following link:
  https://mysagw.local/cases/my-case-id
  
  If you have any questions, please contact us via e-mail (sagw@sagw.ch), phone (+41 31 306 92 50) or online (www.sagw.ch/kontakt-mysagw). Please bear in mind to state the reference number of your application.
  
  Yours sincerely
  
  Your SAGW team
  
  ******
  
  This is an automatically generated message. Please do not reply. The contact details of SASH can be found on the following website:
  www.sagw.ch/kontakt-mysagw
  
  '''
# ---
# name: test_send_work_item_mail[additional-data-False-fr]
  'Nouvelles informations concernant 1984-0023'
# ---
# name: test_send_work_item_mail[additional-data-False-fr].1
  '''
  Bonjour Winston Smith
  
  Nous avons examiné votre demande et avons le plaisir de vous informer que nous avons accordé un crédit-cadre de CHF 23'000.- pour le projet mentionné ci-dessous.
  
  Si nécessaire, vous pouvez télécharger votre demande et la confirmation de l’octroi du financement en format PDF sous la rubrique «Download».
  
  Référence : 1984-0023
  Cliquez sur le lien ci-dessous pour ouvrir votre document:
  https://mysagw.local/cases/my-case-id
  
  Si vous avez des questions, n’hésitez pas à nous contacter à l’adresse sagw@sagw.ch, au numéro de téléphone +41 31 306 92 50 ou via www.sagw.ch/kontakt-mysagw. Nous vous prions de toujours indiquer le numéro de référence lorsque vous posez une question.
  
  Avec nos meilleures salutations,
  
  Votre équipe de l’ASSH
  
  ******
  
  Ceci est un message électronique généré automatiquement, merci de ne pas y répondre. Vous trouverez les coordonnées de l'ASSH sur la page suivante :
  www.sagw.ch/kontakt-mysagw
  
  '''
# ---
# name: test_send_work_item_mail[additional-data-True-de]
  'Neue Informationen zu 1984-0023'
# ---
# name: test_send_work_item_mail[additional-data-True-de].1
  '''
  Guten Tag Winston Smith
  
  Wir benötigen noch weitere Angaben zum Gesuch:
  
  Referenz: 1984-0023
  Über den nachfolgenden Link können Sie direkt auf Ihr Dokument zugreifen, wo Sie die Angaben zu den noch benötigten Informationen finden:
  https://mysagw.local/cases/my-case-id
  
  Sollten Sie Fragen haben, können Sie uns gerne über sagw@sagw.ch, die Telefonnummer +41 31 306 92 50 oder unter www.sagw.ch/kontakt-mysagw kontaktieren. Gerne bitten wir Sie, bei einer Frage immer die Referenznummer anzugeben.
  
  Freundliche Grüsse
  
  Ihr SAGW-Team
  
  ******
  
  Dies ist eine automatisch generierte Email, bitte antworten Sie nicht auf diese Nachricht. Die Kontaktangaben der SAGW können Sie der nachfolgenden Seite entnehmen:
  www.sagw.ch/kontakt-mysagw
  
  '''
# ---
# name: test_send_work_item_mail[additional-data-True-en]
  'New information concerning 1984-0023'
# ---
# name: test_send_work_item_mail[additional-data-True-en].1
  '''
  Dear Winston Smith
  
  Please note that further information is required for the following application:
  
  Reference: 1984-0023
  You can access your document via the following link:
  https://mysagw.local/cases/my-case-id
  
  If you have any questions, please contact us via e-mail (sagw@sagw.ch), phone (+41 31 306 92 50) or online (www.sagw.ch/kontakt-mysagw). Please bear in mind to state the reference number of your application.
  
  Yours sincerely
  
  Your SAGW team
  
  ******
  
  This is an automatically generated message. Please do not reply. The contact details of SASH can be found on the following website:
  www.sagw.ch/kontakt-mysagw
  
  '''
# ---
# name: test_send_work_item_mail[additional-data-True-fr]
  'Nouvelles informations concernant 1984-0023'
# ---
# name: test_send_work_item_mail[additional-data-True-fr].1
  '''
  Bonjour Winston Smith
  
  Nous avons besoin d’informations supplémentaires sur la requête suivante:
  
  Référence : 1984-0023
  Le lien ci-dessous vous permet d’accéder directement à votre document, où vous trouverez les indications sur les informations à fournir:
  https://mysagw.local/cases/my-case-id
  
  Si vous avez des questions, n’hésitez pas à nous contacter à l’adresse sagw@sagw.ch, au numéro de téléphone +41 31 306 92 50 ou via www.sagw.ch/kontakt-mysagw. Nous vous prions de toujours indiquer le numéro de référence lorsque vous posez une question.
  
  Avec nos meilleures salutations,
  
  Votre équipe de l’ASSH
  
  ******
  
  Ceci est un message électronique généré automatiquement, merci de ne pas y répondre. Vous trouverez les coordonnées de l'ASSH sur la page suivante :
  www.sagw.ch/kontakt-mysagw
  
  '''
# ---
# name: test_send_work_item_mail[complete-document-False-de]
  'Neue Informationen zu 1984-0023'
# ---
# name: test_send_work_item_mail[complete-document-False-de].1
  '''
  Guten Tag Winston Smith
  
  Wir haben die Abrechnung des nachfolgenden Gesuchs geprüft und können Ihnen nun in den kommenden Tagen folgenden Beitrag auf die uns vorliegende Zahlungsverbindung überweisen:
  
  CHF 23'000.-
  
  Referenz: 1984-0023
  Über den nachfolgenden Link können Sie direkt auf Ihr Dokument zugreifen, wo Sie die Angaben zu den noch benötigten Informationen finden:
  https://mysagw.local/cases/my-case-id
  
  Sollten Sie Fragen haben, können Sie uns gerne über sagw@sagw.ch, die Telefonnummer +41 31 306 92 50 oder unter www.sagw.ch/kontakt-mysagw kontaktieren. Gerne bitten wir Sie, bei einer Frage immer die Referenznummer anzugeben.
  
  Freundliche Grüsse
  
  Ihr SAGW-Team
  
  ******
  
  Dies ist eine automatisch generierte Email, bitte antworten Sie nicht auf diese Nachricht. Die Kontaktangaben der SAGW können Sie der nachfolgenden Seite entnehmen:
  www.sagw.ch/kontakt-mysagw
  
  '''
# ---
# name: test_send_work_item_mail[complete-document-False-en]
  'New information concerning 1984-0023'
# ---
# name: test_send_work_item_mail[complete-document-False-en].1
  '''
  Dear Winston Smith
  
  After having verified the billing of the application mentioned below, we can confirm that the following amount will be transferred according to the payment details which have been provided:
  
  CHF 23'000.-
  
  Reference: 1984-0023
  You can access your document via the following link:
  https://mysagw.local/cases/my-case-id
  
  If you have any questions, please contact us via e-mail (sagw@sagw.ch), phone (+41 31 306 92 50) or online (www.sagw.ch/kontakt-mysagw). Please bear in mind to state the reference number of your application.
  
  Yours sincerely
  
  Your SAGW team
  
  ******
  
  This is an automatically generated message. Please do not reply. The contact details of SASH can be found on the following website:
  www.sagw.ch/kontakt-mysagw
  
  '''
# ---
# name: test_send_work_item_mail[complete-document-False-fr]
  'Nouvelles informations concernant 1984-0023'
# ---
# name: test_send_work_item_mail[complete-document-False-fr].1
  '''
  Bonjour Winston Smith
  
  Nous avons vérifié le décompte de la demande dont la référence figure ci-dessous. Nous verserons dans les prochains jours, sur les coordonnées de paiement dont nous disposons, le montant suivant:
  
  CHF 23'000.-
  
  Référence : 1984-0023
  Vous pouvez accéder directement à votre document en cliquant sur le lien suivant:
  https://mysagw.local/cases/my-case-id
  
  Si vous avez des questions, n’hésitez pas à nous contacter à l’adresse sagw@sagw.ch, au numéro de téléphone +41 31 306 92 50 ou via www.sagw.ch/kontakt-mysagw. Nous vous prions de toujours indiquer le numéro de référence lorsque vous posez une question.
  
  Avec nos meilleures salutations,
  
  Votre équipe de l’ASSH
  
  ******
  
  Ceci est un message électronique généré automatiquement, merci de ne pas y répondre. Vous trouverez les coordonnées de l'ASSH sur la page suivante :
  www.sagw.ch/kontakt-mysagw
  
  '''
# ---
# name: test_send_work_item_mail[decision-and-credit-False-de]
  'Neue Informationen zu 1984-0023'
# ---
# name: test_send_work_item_mail[decision-and-credit-False-de].1
  '''
  Guten Tag Winston Smith
  
  Die Schweizerische Akademie der Geistes- und Sozialwissenschaften (SAGW) hat Ihr unten genanntes Gesuch geprüft. Leider müssen wir Ihnen mitteilen, dass wir Ihrem Antrag nicht entsprechen können, da weder die formellen noch die materiellen Voraussetzungen für eine Unterstützung erfüllt sind.
  
  Referenz: 1984-0023
  Über den nachfolgenden Link können Sie direkt auf Ihr Dokument zugreifen, wo Sie die Angaben zu den noch benötigten Informationen finden:
  https://mysagw.local/cases/my-case-id
  
  Sollten Sie Fragen haben, können Sie uns gerne über sagw@sagw.ch, die Telefonnummer +41 31 306 92 50 oder unter www.sagw.ch/kontakt-mysagw kontaktieren. Gerne bitten wir Sie, bei einer Frage immer die Referenznummer anzugeben.
  
  Freundliche Grüsse
  
  Ihr SAGW-Team
  
  ******
  
  Dies ist eine automatisch generierte Email, bitte antworten Sie nicht auf diese Nachricht. Die Kontaktangaben der SAGW können Sie der nachfolgenden Seite entnehmen:
  www.sagw.ch/kontakt-mysagw
  
  '''
# ---
# name: test_send_work_item_mail[decision-and-credit-False-en]
  'New information concerning 1984-0023'
# ---
# name: test_send_work_item_mail[decision-and-credit-False-en].1
  '''
  Dear Winston Smith
  
  The Swiss Academy of Humanities and Social Sciences (SAGW/SAHS) has reviewed your request. Unfortunately, we have to inform you that your request cannot be granted since neither the formal nor the material criteria are being fulfilled.
  
  Reference: 1984-0023
  You can access your document via the following link:
  https://mysagw.local/cases/my-case-id
  
  If you have any questions, please contact us via e-mail (sagw@sagw.ch), phone (+41 31 306 92 50) or online (www.sagw.ch/kontakt-mysagw). Please bear in mind to state the reference number of your application.
  
  Yours sincerely
  
  Your SAGW team
  
  ******
  
  This is an automatically generated message. Please do not reply. The contact details of SASH can be found on the following website:
  www.sagw.ch/kontakt-mysagw
  
  '''
# ---
# name: test_send_work_item_mail[decision-and-credit-False-fr]
  'Nouvelles informations concernant 1984-0023'
# ---
# name: test_send_work_item_mail[decision-and-credit-False-fr].1
  '''
  Bonjour Winston Smith
  
  L’Académie suisse des sciences humaines et sociales (ASSH) a examiné votre requête susmentionnée. Malheureusement, nous devons vous informer que nous ne pouvons pas donner suite à votre demande, car ni les conditions formelles ni les conditions matérielles pour un soutien ne sont remplies.
  
  Référence : 1984-0023
  Cliquez sur le lien ci-dessous pour ouvrir votre document:
  https://mysagw.local/cases/my-case-id
  
  Si vous avez des questions, n’hésitez pas à nous contacter à l’adresse sagw@sagw.ch, au numéro de téléphone +41 31 306 92 50 ou via www.sagw.ch/kontakt-mysagw. Nous vous prions de toujours indiquer le numéro de référence lorsque vous posez une question.
  
  Avec nos meilleures salutations,
  
  Votre équipe de l’ASSH
  
  ******
  
  Ceci est un message électronique généré automatiquement, merci de ne pas y répondre. Vous trouverez les coordonnées de l'ASSH sur la page suivante :
  www.sagw.ch/kontakt-mysagw
  
  '''
# ---
# name: test_send_work_item_mail[revise-document-False-de]
  'Neue Informationen zu 1984-0023'
# ---
# name: test_send_work_item_mail[revise-document-False-de].1
  '''
  Guten Tag Winston Smith
  
  Wir benötigen noch weitere Angaben zum Gesuch:
  
  Referenz: 1984-0023
  Über den nachfolgenden Link können Sie direkt auf Ihr Dokument zugreifen, wo Sie die Angaben zu den noch benötigten Informationen finden:
  https://mysagw.local/cases/my-case-id
  
  Sollten Sie Fragen haben, können Sie uns gerne über sagw@sagw.ch, die Telefonnummer +41 31 306 92 50 oder unter www.sagw.ch/kontakt-mysagw kontaktieren. Gerne bitten wir Sie, bei einer Frage immer die Referenznummer anzugeben.
  
  Freundliche Grüsse
  
  Ihr SAGW-Team
  
  ******
  
  Dies ist eine automatisch generierte Email, bitte antworten Sie nicht auf diese Nachricht. Die Kontaktangaben der SAGW können Sie der nachfolgenden Seite entnehmen:
  www.sagw.ch/kontakt-mysagw
  
  '''
# ---
# name: test_send_work_item_mail[revise-document-False-en]
  'New information concerning 1984-0023'
# ---
# name: test_send_work_item_mail[revise-document-False-en].1
  '''
  Dear Winston Smith
  
  Please note that further information is required for the following application:
  
  Reference: 1984-0023
  You can access your document via the following link:
  https://mysagw.local/cases/my-case-id
  
  If you have any questions, please contact us via e-mail (sagw@sagw.ch), phone (+41 31 306 92 50) or online (www.sagw.ch/kontakt-mysagw). Please bear in mind to state the reference number of your application.
  
  Yours sincerely
  
  Your SAGW team
  
  ******
  
  This is an automatically generated message. Please do not reply. The contact details of SASH can be found on the following website:
  www.sagw.ch/kontakt-mysagw
  
  '''
# ---
# name: test_send_work_item_mail[revise-document-False-fr]
  'Nouvelles informations concernant 1984-0023'
# ---
# name: test_send_work_item_mail[revise-document-False-fr].1
  '''
  Bonjour Winston Smith
  
  Nous avons besoin d’informations supplémentaires sur la requête suivante:
  
  Référence : 1984-0023
  Le lien ci-dessous vous permet d’accéder directement à votre document, où vous trouverez les indications sur les informations à fournir:
  https://mysagw.local/cases/my-case-id
  
  Si vous avez des questions, n’hésitez pas à nous contacter à l’adresse sagw@sagw.ch, au numéro de téléphone +41 31 306 92 50 ou via www.sagw.ch/kontakt-mysagw. Nous vous prions de toujours indiquer le numéro de référence lorsque vous posez une question.
  
  Avec nos meilleures salutations,
  
  Votre équipe de l’ASSH
  
  ******
  
  Ceci est un message électronique généré automatiquement, merci de ne pas y répondre. Vous trouverez les coordonnées de l'ASSH sur la page suivante :
  www.sagw.ch/kontakt-mysagw
  
  '''
# ---
